kubeyaml.py

```diff
@@ -193,27 +196,56 @@ def get_image(values):
     return containers
 
 def set_fluxhelmrelease_container(manifest, container, replace):
+    # The logic within this method (almost) equals:
+    # https://github.com/weaveworks/flux/blob/5b15a94397d58b69a2daedae3bcc377e4901435b/image/image.go#L136
+    def parse_ref():
+        reg, im, tag = '', '', ''
+        try:
+            segments = replace.split('/')
+            if len(segments) == 1:
+                im = replace
+            elif len(segments) == 2:
+                domainComponent = '([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])'
+                domain = 'localhost|(%s([.]%s)+)(:[0-9]+)?' % (domainComponent, domainComponent)
+                if re.fullmatch(domain, segments[0]):
+                    reg = segments[0]
+                    im = segments[1]
+                else:
+                    im = replace
+            else:
+                reg = segments[0]
+                im = '/'.join(segments[1:])
+
+            segments = im.split(':')
+            if len(segments) == 2:
+                im, tag = segments
+            elif len(segments) == 3:
+                im = ':'.join(segments[:2])
+                tag = segments[2]
+        except ValueError:
+            pass
+        return reg, im, tag
+
     def set_image(values):
         image = values['image']
         imageKey = 'image'
 
-        if isinstance(image, collections.Mapping) and 'repository' in image and 'tag' in image:
+        if isinstance(image, collections.Mapping) and 'repository' in image:
```

Comment:
I'm not sure the logic below lines up with the formulation of the tests. For one thing: this admits the format

```
values:
  image:
    repository: foo:v3
```

but this is not present in the tests.